pkg/inventory/containers_test.go

```diff
@@ -400,10 +410,134 @@ func TestGetContainersFromPods(t *testing.T) {
+							},
+						},
+						Status: v1.PodStatus{
+							ContainerStatuses: []v1.ContainerStatus{
+								{
+									Name:        "test-container2",
+									Image:       "sha256:20b332c9a70d8516d849d1ac23eff5800cbb2f263d379f0ec11ee908db6b25a8",
+									ImageID:     "docker-pullable://kubernetesui/dashboard@sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93",
+									ContainerID: "docker://a9cd75ad99dd4363bbd882b40e753b58c62bfd7b03cabeb764c1dac97568ad26",
+								},
+							},
+							Phase: v1.PodRunning,
+						},
+					},
+				},
+				ignoreNotRunning:        false,
+				missingRegistryOverride: "custom.registry.io",
+				missingTagPolicy:        "digest",
+				dummyTag:                "",
+			},
+			want: []Container{
+				{
+					Name:        "test-container",
+					ImageTag:    "custom.registry.io/reponame/myimage:0.0.1",
+					ImageDigest: "sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93",
+					ID:          "docker://a9cd75ad99dd4363bbd882b40e753b58c62bfd7b03cabeb764c1dac97568ad26",
+				},
+				{
+					Name:        "test-container2",
+					ImageTag:    "docker.io/kubernetesui/dashboard:v2.7.0",
+					ImageDigest: "sha256:2e500d29e9d5f4a086b908eb8dfe7ecac57d2ab09d65b24f588b1d449841ef93",
+					ID:          "docker://a9cd75ad99dd4363bbd882b40e753b58c62bfd7b03cabeb764c1dac97568ad26",
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := GetContainersFromPods(
+				tt.args.pods,
+				tt.args.ignoreNotRunning,
+				tt.args.missingRegistryOverride,
+				tt.args.missingTagPolicy,
+				tt.args.dummyTag,
+			)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func Test_missingRegistryProcessing(t *testing.T) {
+	type args struct {
+		imageTag                string
+		missingRegistryOverride string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "successfully returns image tag with overridden registry",
+			args: args{
+				imageTag:                "reponame/myimage:0.0.1",
+				missingRegistryOverride: "custom.registry.io",
+			},
+			want: "custom.registry.io/reponame/myimage:0.0.1",
+		},
+		{
+			name: "successfully returns valid image tag without overridden registry",
+			args: args{
+				imageTag:                "docker.io/reponame/myimage:0.0.1",
+				missingRegistryOverride: "custom.registry.io",
+			},
+			want: "docker.io/reponame/myimage:0.0.1",
+		},
+		{
+			name: "successfully returns valid image tag without overridden registry (includes library)",
+			args: args{
+				imageTag:                "docker.io/library/reponame/myimage:0.0.1",
+				missingRegistryOverride: "custom.registry.io",
+			},
+			want: "docker.io/library/reponame/myimage:0.0.1",
+		},
+		{
+			name: "successfully returns valid image tag no repo",
+			args: args{
+				imageTag:                "myimage:0.0.1",
+				missingRegistryOverride: "custom.registry.io",
+			},
+			want: "custom.registry.io/myimage:0.0.1",
+		},
```

Comment:
A test with custom registry and `library/reponame/myimage:0.0.1` as input? To make that example obvious?